### PR TITLE
Vet clinic database: query and update animals table

### PR DESCRIPTION
**Pull request by Juanse7793**
## Requirements

- Add a column species of type string to your animals table. Modify your schema.sql file. :+1: 
- Insert the following data: :heavy_check_mark: 
  - Animal: His name is Charmander. He was born on Feb 8th, 2020, and currently weighs -11kg. He is not neutered and he has never tried to escape.
  - Animal: Her name is Plantmon. She was born on Nov 15th, 2021, and currently weighs -5.7kg. She is neutered and she has tried to escape 2 times.
  - Animal: His name is Squirtle. He was born on Apr 2nd, 1993, and currently weighs -12.13kg. He was not neutered and he has tried to escape 3 times.
  - Animal: His name is Angemon. He was born on Jun 12th, 2005, and currently weighs -45kg. He is neutered and he has tried to escape once.
  - Animal: His name is Boarmon. He was born on Jun 7th, 2005, and currently weighs 20.4kg. He is neutered and he has tried to escape 7 times.
  - Animal: Her name is Blossom. She was born on Oct 13th, 1998, and currently weighs 17kg. She is neutered and she has tried to escape 3 times.
  - Animal: His name is Ditto. He was born on May 14th, 2022, and currently weighs 22kg. He is neutered and he has tried to escape 4 times.

- Inside a transaction update the animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that species columns went back to the state before transaction.
- Inside a transaction: :100: 
  - Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
  - Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
  - Commit the transaction.
  - Verify that change was made and persists after commit.

- Inside a transaction delete all records in the animals table, then roll back the transaction.
After the roll back verify if all records in the animals table still exist. :+1: 
- Inside a transaction: :nerd_face: 
  - Delete all animals born after Jan 1st, 2022.
  - Create a savepoint for the transaction.
  - Update all animals' weight to be their weight multiplied by -1.
  - Rollback to the savepoint
  - Update all animals' weights that are negative to be their weight multiplied by -1.
  - Commit transaction
- Write queries to answer the following questions: :heavy_check_mark: 
  - How many animals are there?
  - How many animals have never tried to escape?
  - What is the average weight of animals?
  - Who escapes the most, neutered or not neutered animals?
  - What is the minimum and maximum weight of each type of animal?
  - What is the average number of escape attempts per animal type of those born between 1990 and 2000?

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -8,3 +8,9 @@ CREATE TABLE animals(
    neutered         BOOLEAN,
    weight_kg       DECIMAL
 );
+
+ALTER TABLE animals ADD species text NULL;
+
+BEGIN
+ALTER TABLE animals RENAME COLUMN species TO unspecified;
+ROLLBACK
```
Comment (KDlamini) [inline schema.sql]: This is a query, therefore it should be in the `queries.sql` file. Everything that is a query that helps to analyze data should be saved in the queries.sql file. 

- Kindly make this correction.

File: queries.sql [hunk 2]
```diff
@@ -8,3 +8,29 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg >= 10.50;
...
 SELECT * From animals WHERE name != 'Gabumon';
 SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+
+UPDATE animals
+SET species = 'digimon' 
+WHERE name LIKE '%mon%';
+
+UPDATE animals
+SET species = 'pokemon' 
+WHERE name NOT LIKE '%mon%';
```
Comment (KDlamini) [inline queries.sql:25]: This query must be performed inside a Transaction. Kindly add `BEGIN` before the update and `COMMIT` after the update to end the transaction.

File: queries.sql [hunk 3]
```diff
@@ -8,3 +8,29 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg >= 10.50;
...
+
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
+
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT SP1;
+UPDATE animals SET weight_kg = weight_kg * -1;
+ROLLBACK TO SAVEPOINT SP1;
```
Comment (KDlamini) [inline queries.sql:36]: After you `ROLLBACK TO SAVEPOINT SP1;` you must "update all animals' weights that are negative to be their weight multiplied by -1." as per project requirements.
image

For example:
```
    UPDATE animals SET weight_kg = weight_kg * -1; 
    ROLLBACK TO SP1;
    UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
```

**Review by KDlamini — CHANGES_REQUESTED** [review]
### Hi @Juanse7793 🙋‍♀️,  

Great work so far! 💯 
There are some issues that you still need to work on improving your project according to the requirements. You are almost there! 💪🏾 

## Required Changes ♻️
 
**Kindly look through the comments below 👇 for required changes**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
N/A

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by keroloussamy — APPROVED** [review]
Hi @Juanse7793, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To highlight 🎯


- Good code design and implementation. ✔️
- Add all the requirements. ✔️

clap


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**